modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1142,10 +1175,23 @@ def _get_dict_of_block_index(self, axis, indices):
                         )
                         dict_of_slices.update({last_part: slice(None, last_idx[0])})
                         return dict_of_slices
-        # Sort and convert negative indices to positive
-        indices = np.sort(
-            [i if i >= 0 else max(0, len(self.axes[axis]) + i) for i in indices]
-        )
+        if isinstance(indices, list):
+            # Converting python list to numpy for faster processing
+            indices = np.array(indices, dtype=np.int64)
+        negative_mask = np.less(indices, 0)
+        has_negative = np.any(negative_mask)
+        if has_negative:
+            # We're going to modify 'indices' inplace in a numpy way, so doing a copy/converting indices to numpy.
+            indices = (
+                indices.copy()
+                if isinstance(indices, np.ndarray)
+                else np.array(indices, dtype=np.int64)
```

Comment:
We already convert `indices` list to np.array above.

Reply:
The conversion to a numpy array above is done for python lists only, this was done due to performance reasons, as it's known for sure, that the python list is slower in numpy vectorized operations, at the same time, performance with `pandas.Index` (and probably for some other containers) is good, so we don't want to convert them to numpy.

The conversion in the commented lines is done to convert possibly immutable containers (like `pandas.Index`) to a mutable numpy array.

Reply:
ok